realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java

```diff
@@ -115,6 +116,86 @@ public void create_duplicatedNameThrows() {
     }
 
     @Test
+    public void createWithPrimaryKeyField_string() {
+        // Not nullable
+        realmSchema.createWithPrimaryKeyField("FooNonNull", "pkField", String.class, FieldAttribute.REQUIRED);
+        RealmObjectSchema objectSchema = realmSchema.getSchemaForClass("FooNonNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.STRING, objectSchema.getFieldType("pkField"));
+        assertFalse(objectSchema.isNullable("pkField"));
+
+        // Nullable
+        realmSchema.createWithPrimaryKeyField("FooNull", "pkField", String.class);
+        objectSchema = realmSchema.getSchemaForClass("FooNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.STRING, objectSchema.getFieldType("pkField"));
+        assertTrue(objectSchema.isNullable("pkField"));
+    }
+
+    @Test
+    public void createWithPrimaryKeyField_integer() {
+        // Not nullable
+        realmSchema.createWithPrimaryKeyField("FooNonNull", "pkField", int.class);
```

Comment:
you comments lead me to the same stupid bug twice!! thanks a lot! https://github.com/realm/realm-java/pull/4884/commits/5bd9b2b7c4667256cac9030c7208006fddc7a16d